diagonalLines.py

```diff
@@ -0,0 +1,125 @@
+    point_set = set()
+    lines = []
+    neg = True
+    #repeats ignoring duplicate points
+    while len(lines) < num_lines:
+        #get a random even number for x adn y
+        x = random.randint(0, sections // 2 - 1) * 2
+        y = random.randint(0, x // 2) * 2
+
+        #calculate the maximum length
+        top = sections - x - y if x == y and x == 2 else sections-x
+        #pick a random length between 1 and the maximum
+        length = random.randint(1, top)
+        #alternate the positivity of the gradient depending on starting point
+        pos = (x in (2, 4)and y in (2, 4))
+
+        if (x, y) not in point_set:
+            lines.append((x, y, length, pos))
+            point_set.add((x, y))
+    return lines
+
+def make_tiles(size, tiles ,image_size, sw=1):
+    #takes number of sections in each quadrant (indicates intricacy)
+    #tiles is the number of repetitions across and down
+    #imagesize is number of pixels across and down but it also affects
+    #the relative stroke width --> sw is stroke width
+    image_pad = 50
+    header = f'<svg viewBox="-{image_pad} -{image_pad} {image_size + 2 * image_pad} {image_size + 2 * image_pad}" xmlns="http://www.w3.org/2000/svg">\n'
+    print(header)
+    tile_size = image_size / tiles
+    padding = image_size / tiles / 2
+    padding = 40
+    sections = size
+    for row in range(tiles):
+        for column in range(tiles):
+            #cutting off at the border is not working
+
+            #make a tile quadrant group (without <g> wrapping just yet)
+            group = ''
+            top_left_x = column * tile_size + padding / 2
+            top_left_y = row * tile_size + padding / 2
+            #print(top_left_x, top_left_y)
+            square_size = ((tile_size - padding) / size) / 2
+            lines= create_lines(sections)
+            for line in lines:
+                #do the same thing (ish) in the four different quadrants
+                x, y, length, pos = line
+                x1 = x
+                y1 = y
+                
+                #print(x1,y1,length,pos,a,b)
+                l = draw_line(x1, y1, length, pos, square_size, sections, top_left_x, top_left_y, sw)
+                group += l +'\n'
+                #x, y, length, pos, size, sections, top_left_x, top_left_y, sw, sym=False, color='black'
+                if x != y or pos:
+                    x1 = y
+                    y1 = x
+                    l2 = draw_line(x1, y1, -length, pos, square_size, sections, top_left_x, top_left_y, sw, sym = True)
+                    #x, y, length, pos, size, sections, top_left_x, top_left_y, sw, sym=False, color='black'
+                    group += l2 +'\n'
+                    
+            #repeat this quadrant 3 more times
+            for quad in range(4):
+                #convert quad to binary for positioning
+                rotation = quad * 90
+                rotationX = top_left_x + (tile_size - padding) / 2
```

Comment:
`rotationX` and `rotationY` are always the _same_ value. Put them outside the loop?